packages/jest-haste-map/src/watchers/FSEventsWatcher.ts

```diff
@@ -130,13 +130,12 @@ class FSEventsWatcher extends EventEmitter {
   /**
    * End watching.
    */
-  close(callback?: () => void): void {
-    this.fsEventsWatchStopper().then(() => {
-      this.removeAllListeners();
-      if (typeof callback === 'function') {
-        process.nextTick(callback.bind(null, null, true));
-      }
-    });
+  async close(callback?: () => void): Promise<void> {
+    await this.fsEventsWatchStopper();
```

Comment:
an actual code change, but we do wait for the promise instead of using the callback, so probably a bug fix: https://github.com/facebook/jest/blob/b51564b072f89c2aa8cb2a1dd26ab54bc5c1c146/packages/jest-haste-map/src/index.ts#L1046